src/main/java/net/snowflake/client/core/SFBaseStatement.java

```diff
@@ -39,9 +39,11 @@ public abstract class SFBaseStatement {
   public void addProperty(String propertyName, Object propertyValue) throws SFException {
     statementParametersMap.put(propertyName, propertyValue);
 
-    // for query timeout, we implement it on client side for now
     if ("query_timeout".equalsIgnoreCase(propertyName)) {
+      // Client side implementation
       queryTimeout = (Integer) propertyValue;
+      // Set server parameter for supporting query timeout on async queries
+      statementParametersMap.put("STATEMENT_TIMEOUT_IN_SECONDS", (Integer) propertyValue);
```

Comment:
I am also not sure if we want to mix the behaviour here for sync and async queries - do our users receive exactly the same error as before when driver internall was timing out?
To play safe I would suggest creating new method in SnowflakeStatement with explicit name `setAsyncQueryTimeout`
in the existing `setQueryTimeout` method I would guard the `STATEMENT_TIMEOUT_IN_SECONDS ` with some new driver parameter  with default to false e.g. SUPPORT_IMPLICIT_ASYNC_QUERY_TIMEOUT
@sfc-gh-abenzaoui WDYT?

Reply:
I've updated it to use a new connection property 'SUPPORT_IMPLICIT_ASYNC_QUERY_TIMEOUT' within the existing `setQueryTimeout` method. I can add switch the implementation to use a new `setAsyncQueryTimeout` method, but it feels unnecessary if we're already hiding it behind the connection parameter. Is this approach acceptable?

Reply:
I mean in my comment that let's create a special method on SnowflakeStatement just to set the async query timeout for the single statment
I think we need such methods to explicitly set timeout for async query to call without having to explicitly change connection parameter